src/controller/recipe/mod.rs

```diff
@@ -62,6 +62,7 @@ fn hash_query(q: &SqlQuery) -> QueryID {
     h.finish()
 }
 
+#[allow(unused)]
```

Comment:
I'm not sure how this is unused -- `self.recipe` on `ControllerInner` stores a `Recipe`, so it should use this type?